packages/components/src/vue/vuex-modules/data/lexis.js

```diff
@@ -200,7 +200,10 @@ export default class Lexis extends Module {
     }
     if (!wordUsageExamples) { wordUsageExamples = this._appApi.getWordUsageExamplesQueryParams(textSelector) }
 
-    this._appApi.newLexicalRequest(textSelector.normalizedText, textSelector.languageID, textSelector.data, source)
+    if (!forWordlist) {
+      this._appApi.newLexicalRequest(textSelector.normalizedText, textSelector.languageID, textSelector.data, source)
+    }
```

Comment:
I'm wondering if there are going to be any unexpected side effects of issuing the lexicalQuery.getData request without going through the regular process of initializing a new lexical request. getData issues a number of events that affect the interface.  It might be that we need to do more here, and suppress events during the lexical query getData when we're doing this sort of background request. I'd like to get some thoughts from @kirlat on this.

Reply:
From my point of view `newLexicalRequest` method change data in ui, app, storage modules.
It is not needed for background process of updating homonym.
I belive that we need only to update homonym object here - that's why I think it is ok.
Would be glad to @kirlat 's opinion.

Reply:
@irina060981 I've tried to simplify a number of options of lexical query methods and I think it is important in order to keep it maintainable. Right now we have two types of requests identified by a `source` prop: a get selected text request from a page (identified by a `page` source) and a one initiated by the lookup component (source is `lookup`). It looks to me (please correct me if I'm wrong) that this is a third type of request. So maybe we can use a `source` filed to identify it (i.e. having it set to something like `wordList`)? We can modify a `lookupText()` API method so that it will take `source` as an additional parameter and will pass it through to `lexicalQuery`. Or we can create another special method like `wordListLookup()` that will be similar to `lookupText()` but will use a source of `wordList`. What do you think about that?